enhancements/machine-api/control-plane-machine-set.md

```diff
@@ -248,6 +247,73 @@ type ControlPlaneMachineSetStatus struct {
     // +optional
     Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
+
+// FailureDomain represents the different configurations required to spread Machines
+// across failure domains on different platforms.
+// +union
+type FailureDomain struct {
+  // Platform identifies the platform for which the FailureDomain represents
+  // +kubebuilder:validation:Enum:="aws";"azure";"gcp";"openstack"
+  // +unionDiscriminator
+  // +optional
+  Platform configv1.PlatformType `json:platform,omitempty`
```

Comment:
I realise this forms part of a union discriminator pattern (although I couldn't find exactly where that's defined), but it seems like a strangely high level value to be buried so deeply in this structure.

Reply:
You might find [this doc](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1027-api-unions/README.md) helpful as it talks about how they work

This field is optional, and I don't expect most users to set it. The value of this field, IIUC, must match the json name of the struct that is non nil within the union, but as long as only one struct is non-nil, it's not required.
This field is purely informational as to which struct in the union is intended to be used